### PR TITLE
[Tabletop Game] Mafia

### PR DESCRIPTION
**Pull request by PuroSlavKing**
## About the PR

> bruh

The game is for 10 players, contains role cards:
2 - Mafia _(Syndie)_
1 - Detective
1 - Medic 
6 - Passenger

don't merge without: https://github.com/space-wizards/space-station-14/pull/15409

**Media**

https://user-images.githubusercontent.com/103608145/232048093-7c530029-fae9-4a6b-be15-a97696c313c8.mp4

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: PuroSlavKing
- add: Added new tabletop game - card mafia!

### CODE REVIEW

**Comment by deltanedas** [comment]
among us? in my spaceman game???

File: Resources/Prototypes/Entities/Objects/Fun/Tabletop/mafia.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    - type: StorageVisualizer
+      state_open: carddetective
+      state_closed: cardunknown
+
+- type: entity
+  name: card
+  suffix: Passenger
+  parent: BaseItem
+  id: CardPassenger
+  description: Turn over the card to find out your role.
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Fun/Tabletop/mafia.yml:91]: description, sprite, item, storage, userinterface can be put into an abstract mafia card prototype which each one extends

File: Resources/Prototypes/Entities/Objects/Fun/Tabletop/mafia.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    drawdepth: SmallObjects
+    layers:
+    - state: cardunknown
+      map: ["enum.StorageVisualLayers.Door"]
+    netsync: false
+  - type: Item
+    sprite: Objects\Fun\Tabletop\mafia\mafiacards.rsi
+    size: 1
+  - type: Storage
+    capacity: -1
```
Comment (deltanedas) [inline Resources/Prototypes/Entities/Objects/Fun/Tabletop/mafia.yml:104]: this is a funny hack, is there any other existing way to visualize it?

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by deltanedas** [comment]
also would be cool if it had a manual item in the box since i doubt many people have played mafia

File: Resources/Prototypes/Entities/Objects/Fun/Tabletop/mafia.yml (hunk 2 above)
Comment (PuroSlavKing) [reply Resources/Prototypes/Entities/Objects/Fun/Tabletop/mafia.yml:104]: > this is a funny hack, is there any other existing way to visualize it?

I failed, at best it stopped spawning, and at worst it restarting the server.

Comment (PuroSlavKing) [reply Resources/Prototypes/Entities/Objects/Fun/Tabletop/mafia.yml:104]: > as in removing all the storage visualizer/ui related stuff and using something intended for flipping cards

I didn't find anything like that, the closest was SpawnOnUse, but the cards stopped spawning. To make everything beautiful, you need to create a new interface via #C, but this is beyond my capabilities.

Comment (deltanedas) [reply Resources/Prototypes/Entities/Objects/Fun/Tabletop/mafia.yml:104]: ok keep the funny hack for now i guess, if it still exists when merged i will make write a visualizer to toggle sprite state

**Comment by deltanedas** [comment]
i will also make thing to pick a random card from deck

**Comment by deltanedas** [comment]
keep in mind card picking wont work until pr merged

**Comment by deltanedas** [comment]
you can merge master now

**Comment by PuroSlavKing** [comment]
> you can merge master now

image
image

**Comment by deltanedas** [comment]
indent it to be inside the component

**Pull request closed without merging**